test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
+                           [1.0  0.0  0.0; 1.0  1.0  0.0;  0.0  1.0  0.0]
+    @test Array(spdiagm(ones(2), -1, 3, 3)) == diagm(ones(2), -1)
+end
 
+@testset "Issue #4986, reinterpret" begin
+    sfe22 = speye(Float64, 2)
+    mfe22 = eye(Float64, 2)
+    @test reinterpret(Int64, sfe22) == reinterpret(Int64, mfe22)
+end
 
-# issue #5386
-K,J,V = findnz(SparseMatrixCSC(2,1,[1,3],[1,2],[1.0,0.0]))
-@test length(K) == length(J) == length(V) == 1
+@testset "Issue #5190" begin
+    @test_throws ArgumentError sparsevec([3,5,7],[0.1,0.0,3.2],4)
+end
 
-# https://groups.google.com/d/msg/julia-users/Yq4dh8NOWBQ/GU57L90FZ3EJ
-A = speye(Bool, 5)
-@test find(A) == find(x -> x == true, A) == find(Array(A))
+@testset "Issue #5386" begin
+    K,J,V = findnz(SparseMatrixCSC(2,1,[1,3],[1,2],[1.0,0.0]))
+    @test length(K) == length(J) == length(V) == 1
+end
 
+@testset "https://groups.google.com/d/msg/julia-users/Yq4dh8NOWBQ/GU57L90FZ3EJ" begin
+    A = speye(Bool, 5)
+    @test find(A) == find(x -> x == true, A) == find(Array(A))
+end
 
-# issue #5824
-@test sprand(4,5,0.5).^0 == sparse(ones(4,5))
+@testset "Issue #5824" begin
+    @test sprand(4,5,0.5).^0 == sparse(ones(4,5))
+end
 
-# issue #5985
-@test sprand(Bool, 4, 5, 0.0) == sparse(zeros(Bool, 4, 5))
-@test sprand(Bool, 4, 5, 1.00) == sparse(ones(Bool, 4, 5))
-sprb45nnzs = zeros(5)
-for i=1:5
-    sprb45 = sprand(Bool, 4, 5, 0.5)
-    @test length(sprb45) == 20
-    sprb45nnzs[i] = sum(sprb45)[1]
+@testset "Issue #5985" begin
+    @test sprand(Bool, 4, 5, 0.0) == sparse(zeros(Bool, 4, 5))
+    @test sprand(Bool, 4, 5, 1.00) == sparse(ones(Bool, 4, 5))
+    sprb45nnzs = zeros(5)
+    for i=1:5
+        sprb45 = sprand(Bool, 4, 5, 0.5)
+        @test length(sprb45) == 20
+        sprb45nnzs[i] = sum(sprb45)[1]
+    end
+    @test 4 <= mean(sprb45nnzs) <= 16
 end
-@test 4 <= mean(sprb45nnzs) <= 16
 
-# issue #5853, sparse diff
-for i=1:2, a=Any[[1 2 3], [1 2 3]', eye(3)]
-    @test all(diff(sparse(a),i) == diff(a,i))
+@testset "Issue #5853, sparse diff" begin
+    for i=1:2, a=Any[[1 2 3], [1 2 3]', eye(3)]
+        @test all(diff(sparse(a),i) == diff(a,i))
+    end
 end
 
 # test for "access to undefined error" types that initially allocate elements as #undef
-@test all(sparse(1:2, 1:2, Number[1,2])^2 == sparse(1:2, 1:2, [1,4]))
-sd1 = diff(sparse([1,1,1], [1,2,3], Number[1,2,3]), 1)
-
-# issue #6036
-P = spzeros(Float64, 3, 3)
-for i = 1:3
-    P[i,i] = i
+@testset "Access to undefined error" begin
+    @test all(sparse(1:2, 1:2, Number[1,2])^2 == sparse(1:2, 1:2, [1,4]))
+    sd1 = diff(sparse([1,1,1], [1,2,3], Number[1,2,3]), 1)
 end
 
-@test minimum(P) === 0.0
-@test maximum(P) === 3.0
-@test minimum(-P) === -3.0
-@test maximum(-P) === 0.0
+@testset "Issue #6036" begin
+    P = spzeros(Float64, 3, 3)
+    for i = 1:3
+        P[i,i] = i
+    end
+
+    @test minimum(P) === 0.0
+    @test maximum(P) === 3.0
+    @test minimum(-P) === -3.0
+    @test maximum(-P) === 0.0
```

Comment:
The rest of the information in the associated comment seems critical for this description to make sense?